Gemfile

```diff
@@ -65,6 +65,7 @@ group :test, :development do
   gem 'rack-test', :require => 'rack/test'
   gem 'simplecov', :require => false
   gem 'pry-byebug'
+  gem 'pry-doc'
```

Comment:
@mejackreed's worry about future devs figuring out what relation this has to the commit might be assuaged somewhat if the commit message had more explanation?

i'm a big fan of good commit messages, where there's a concise subject line, and the body has a good detailed summary of the changes (but not verbatim repetition of the diff, in the same way that good code comments should be more of an overview than `a += 2   # adds 2 to a`).

@darrenleeweber, you've left a bunch of helpful explanatory notes as comments on this PR, which i think is much appreciated, but it'd be helpful to future users of the repo (who might be slogging through commits to investigate a bug) if similar helpful info was in the commit message itself.  though it can be a pain, i think it's worth spending some time crafting good commit messages, especially for a change as big as this.  and i've found that it can help me understand the changes i made if i'm forced to summarize them, in the same way that i find it good to be forced to write a quality explanatory comment about a method.